### PR TITLE
Fix macOS deployments only running on newest macOS

### LINKED ISSUE #44: macOS deployments only work on the latest macOS version
The issue is `MACOSX_DEPLOYMENT_TARGET` here: https://github.com/NiklasEi/bevy_game_template/blob/main/.github/workflows/release.yaml#L48
The current config will set this to the newest macOS version, which disables the app on all older machines. 
I'm no expert at what a sensible default would be, but I'd wager a good candidate could be `11.0` Big Sur from November 19, 2020, since it is the first version to support universal binaries.

(credit for this discovery and suggested fix goes to @bash who's sitting next to me 😄)

### PR DESCRIPTION
**Pull request by janhohenheim**
Fixes #44
As described in the issue, this uses Big Sur.

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -18,7 +18,10 @@ permissions:
 jobs:
   build-macOS:
     runs-on: macos-latest
-
+    
+    env: 
+      # macOS 11.0 Big Sur is the newest version to support universal binaries
```
Comment (NiklasEi) [inline .github/workflows/release.yaml:23]: Isn't it the first version to support universal binaries?

**Review by NiklasEi — APPROVED** [review]
Hey, thanks for contributing. I don't have a test setup for macOS, but this makes sense to me :+1:

File: .github/workflows/release.yaml (hunk 1 above)
Comment (janhohenheim) [reply .github/workflows/release.yaml:23]: eeeeh yep 😅

Comment (janhohenheim) [reply .github/workflows/release.yaml]: Fixed; [`310fd7c` (#45)](https://github.com/NiklasEi/bevy_game_template/pull/45/commits/310fd7ca06e77f600766aca9c1f5d31babbf97de)

**Review by NiklasEi — APPROVED** [review]

**Pull request merged**